runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -218,19 +220,55 @@ public TxnContext txn(@Nonnull final String namespace, IsolationLevel isolationL
     }
 
     /**
-     * Return the address of the latest updated made in this table.
+     * Return the address of the latest update made in this table.
+     *
+     * Note: we can't deliberately return the tail of the stream map
+     * as this entry might be a HOLE, we need to filter and return only that
+     * corresponding to the last DATA entry.
      *
      * @param namespace - namespace that this table belongs to.
      * @param tableName - table name of this table without the namespace prefixed in.
      * @return stream tail of this table.
      */
     public long getHighestSequence(@Nonnull final String namespace,
                                    @Nonnull final String tableName) {
-        return this.runtime.getSequencerView().query(
-                CorfuRuntime.getStreamID(
-                        TableRegistry.getFullyQualifiedTableName(namespace, tableName)
-                )
-        );
+
+        // Overall strategy:
+        // (1) Retrieve streams full address map from sequencer
+        // (2) Read entries in reverse to confirm it's a DATA type entry (for optimization
+        // read in batches). A batch size of 4 is guaranteed to suit one edge case case scenario: i.e.,
+        // consecutive checkpoints enforcing holes (3 rounds of CP accumulate before trim happens == 3 holes)
+
+        long startTime = System.nanoTime();
+        StreamAddressSpace streamAddressSpace = this.runtime.getSequencerView()
+                .getStreamAddressSpace(new StreamAddressRange(CorfuRuntime.getStreamID(
+                        TableRegistry.getFullyQualifiedTableName(namespace, tableName)), Address.MAX, Address.NON_ADDRESS));
```

Comment:
Just wondering if it would suffice to just the last getHighestSequenceNumberBatchSize number of entries from Address.MAX, and fetch more if the first batch is depleted, that way we avoid sending large bitmaps over?
(But since this is just the bitmap, perhaps its ok too)

Reply:
It's a valid point, but I honestly doubt this would be necessary as we have never seen these bitmaps being a problem on transfer (and on restart they are completely transferred from log unit to sequencer), also if checkpointing is happening bitmaps are trimmed regularly.